include/irr/builtin/material_compiler/glsl/common.glsl

```diff
@@ -347,6 +330,16 @@ void setCurrInteraction(in vec3 N)
 	irr_glsl_IsotropicViewSurfaceInteraction interaction = irr_glsl_calcFragmentShaderSurfaceInteraction(campos, WorldPos, N);
 	currInteraction = irr_glsl_calcAnisotropicInteraction(interaction);
 }
+void setCurrInteraction(in vec3 N, in bool ts)
+{
+#ifdef NO_TWOSIDED
+	setCurrInteraction(N);
+#else
+	vec3 campos = irr_glsl_MC_getCamPos();
+	vec3 view = campos - WorldPos;
+	setCurrInteraction((ts && dot(N,view)<0.0) ? -N : N);
```

Comment:
`NdotV` should already be precomputed in your current interaction, and thats the one you should be using!

look at the code listing:
https://github.com/buildaworldnet/IrrlichtBAW/pull/424#discussion_r509456818

Reply:
?? i cant read interaction to set interaction

Reply:
also https://github.com/buildaworldnet/IrrlichtBAW/pull/424#discussion_r509643497

Reply:
yes you can read current interaction to flip it

Also the `view` should be cached.. see my comment about `irr_glsl_MC_getNormalizedV()`